### PR TITLE
feat: add DATEADD and DATESUB functions

### PR DESCRIPTION
**Pull request by jzaralim**
### Description 
as the title says. this won't pass until #7733 is merged since that PR contains changes required to enable DATE for UDFs.
### Testing done 
QTT + unit tests.

### Reviewer checklist
- [ ] Ensure docs are updated if necessary. (eg. if a user visible feature is being added or changed).
- [ ] Ensure relevant issues are linked (description should include text like "Fixes #<issue number>")

### CODE REVIEW

**Review by JimGalasyn — APPROVED** [review]
LGTM!

File: ksqldb-functional-tests/src/test/resources/query-validation-tests/dateadd.json [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+      ],
+      "inputs": [
+        {"topic": "test", "key": 0, "value": {"date": 5}},
+        {"topic": "test", "key": 0, "value": {"date": 10}},
+        {"topic": "test", "key": 0, "value": {"date": 2}}
+      ],
+      "outputs": [
+        {"topic": "TEST2", "key": 0, "value": {"VALUE": 0}},
+        {"topic": "TEST2", "key": 0, "value": {"VALUE": 5}},
+        {"topic": "TEST2", "key": 0, "value": {"VALUE": -3}}
```
Comment (spena) [inline ksqldb-functional-tests/src/test/resources/query-validation-tests/dateadd.json:42]: I'm not sure if supporting this negative results. What does a `DATE` data type does when detecting a negative value?

Comment (jzaralim) [reply ksqldb-functional-tests/src/test/resources/query-validation-tests/dateadd.json:42]: Connect translates dates before Epoch to negative numbers, so in this case its `1969-12-28`.

**Review by spena — APPROVED** [review]
LGTM

**Pull request merged**